### PR TITLE
Preview support

### PR DESCRIPTION
**Pull request by amit-jain**
## Description

Support replication on the preview agent

## Related Issue

https://github.com/adobe/aem-test-samples/issues/38

## Motivation and Context

The support for preview agent will enable testing for the preview environments recently added

## How Has This Been Tested?

Changes tested on a cloud environment

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [X] I have added tests to cover my changes.
- [X] All new and existing tests passed.

### CODE REVIEW

**Comment by amit-jain** [comment]
Relies on changes in the PR - https://github.com/adobe/aem-testing-clients/pull/59

Is based on changes in https://github.com/adobe/aem-test-samples/pull/37

**Comment by amit-jain** [comment]
@dulvac Appreciate your review

File: smoke/pom.xml [hunk 1]
```diff
@@ -243,7 +243,7 @@
         <dependency>
             <groupId>com.adobe.cq</groupId>
             <artifactId>aem-cloud-testing-clients</artifactId>
-            <version>0.8.0</version>
+            <version>0.8.1-SNAPSHOT</version>
```
Comment (dulvac) [inline smoke/pom.xml]: This needs to be a released version

File: smoke/src/main/java/com/adobe/cq/cloud/testing/it/smoke/PublishEndToEndIT.java [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    public static CQAuthorPublishClassRule cqBaseClassRule = new CQAuthorPublishClassRule();
+
+    @Rule
+    public CQRule cqBaseRule = new CQRule(cqBaseClassRule.authorRule, cqBaseClassRule.publishRule);
+
+    @Rule
+    public Page root = new Page(cqBaseClassRule.authorRule);
+
+    @Rule
+    public PublishRule publishRule = new PublishRule(root, cqBaseClassRule.authorRule, cqBaseClassRule.publishRule);
```
Comment (dulvac) [inline smoke/src/main/java/com/adobe/cq/cloud/testing/it/smoke/PublishEndToEndIT.java]: Please rename `PublishRule` to something that doesn't clash with `cqBaseClassRule.publishRule` and creates confusion. Like `ContentPublishingRule`

**Review by dulvac — CHANGES_REQUESTED** [review]

**Comment by amit-jain** [comment]
@tmaret Could you please review

**Review by dulvac — APPROVED** [review]

**Comment by amit-jain** [comment]
Merged all changes from PR #37. Also, needs merge and release of https://github.com/adobe/aem-testing-clients/pull/62 before this is merged

**Comment by amit-jain** [comment]
@dulvac This issue is finally updated with the latest released version for aem-testing-clients and good to be merged. Thanks!!

**Pull request closed without merging**